os/kernel/preference/preference_callback.c

```diff
@@ -0,0 +1,218 @@
+/****************************************************************************
+ *
+ * Copyright 2018 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+#include <tinyara/config.h>
+#include <debug.h>
+#include <queue.h>
+#include <errno.h>
+#include <string.h>
+#include <mqueue.h>
+#include <signal.h>
+#include <fcntl.h>
+#include <stdbool.h>
+#include <stdio.h>
+#include <unistd.h>
+#include <tinyara/preference.h>
+
+struct key_cb_list_s {
+	struct key_cb_list_s *flink;
```

Comment:
You using this?

Reply:
yes. it is used for list by key

Reply:
I mean *flink

Reply:
yes. This flink is used in sq_XX operations for a list of keys.
For example, sq_addlast((FAR sq_entry_t *)list, &g_key_cb_list); in preference_register_callback function